source/base/utilities.cc

```diff
@@ -123,6 +124,37 @@ namespace Utilities
     return std::string( input, start_idx, end_idx+1-start_idx);
   }
 
+
+
+  std::string
+  trim_whitespace(const std::string &input)
+  {
+    std::string::size_type left = 0;
+    std::string::size_type right = input.size() - 1;
+
+    std::locale current_locale;
+
+    for (; left < input.size(); ++left)
+      {
+        if (!std::isspace(input[left], current_locale))
```

Comment:
Is that always true? The C++ docs say that the current locale may define extra whitespace characters not in the standard one. I am probably being too conservative here. Would it be better to ignore that problem and just use the `C` locale whitespace characters?